src/containerUtils.js

```diff
@@ -0,0 +1,54 @@
+import invariant from 'invariant';
+import React from 'react';
+import Relay from 'react-relay';
+
+import BatchedRelayContainerBase from './BatchedRelayContainerBase';
+
+export function createBatchedRelayContainer(
+    childElement,
+    params,
+    routeName,
+    queryBatcher
+) {
+  const {queries, fragments} = queryBatcher.flush();
+  const route = {name: routeName, queries, params};
+
+  class BatchedRelayContainer extends BatchedRelayContainerBase {
+    static childElement = childElement;
+    static fragments = fragments;
+  }
+
+  return (
+    <Relay.RootContainer
```

Comment:
It looks like you can't inject additional `props` into the root-most container any more? I guess this is so you can inject them per-route instead, which makes sense. However, the root-most container ought to be where it makes sense to have `onReadyStateChange`, though.